hot-reload-analysis/src/main/kotlin/org/jetbrains/compose/reload/analysis/resolveDirty.kt

```diff
@@ -293,7 +300,7 @@ private fun resolveTransitivelyDirty(
                 ?: current.classIndex[element.memberId.classId]
 
             classInfo?.fields?.forEach { (fieldId, field) ->
-                if (field.isStatic) {
+                if (field.isStatic && field.resourceContentHash == null) {
```

Comment:
Upon first reading, this is not clear to me: 
Can you provide some insights on when this check matters?

Reply:
The code marks all fields dirty if a class initializer changes. If we talk about resource accessors it means:

-- if we change one text resource, the class initializer of a resource accessor class will change (due to internal implementation)
-- the code will mark all resource accessors as dirty even if only one has changed
-- then we invalidate more scopes that it is actually required

Filtering resource accessor fields from the picture fixes the problem.

In general the code is too conservative. Ideally, it is better to analyze which static fileds are actually changed in the changed initializer to mark them dirty (not all of them).

My change is also safe because we already mark changed resource accessors as dirty (and not more).

Reply:
Okay, in this case, would it be possible to move the filtering to 'Phase 1' instead? Similar to how we handle `ComposableSingleton`?